### PR TITLE
Deprecate slave keyword in favor of puppet.

### LINKED ISSUE #7986: master/slave GDScript keyword name choice might be unfortunate
There's been something of a trend towards moving away from “master” and “slave” terminology, because of its associations with, well, slavery: https://en.wikipedia.org/wiki/Master-slave_%28technology%29#Appropriateness_of_usage

Besides that issue, sometimes alternative naming choices are clearer anyway (e.g. “leader/follower”), especially to people from a non-technical background. It's not entirely clear to me from the names what they mean. My first guess was it means an authoritative server (“master”) and replicating client (“slave”) relationship, but I'm not quite sure.

At the risk of starting a flame war, it might be worth changing them. If they were to be changed, better to do it now while they're not in a stable release.

### PR DESCRIPTION
**Pull request by Faless**
After much discussion, the name change is here.
- The `slave` keyword is being renamed to `puppet`
- The `slavesync` keyword is being renamed to `puppetsync`

The `slave` keyword will still exists in `3.1` as deprecated, and will be dropped in future release.

Additionally the `sync` keyword is now more clearly deprecated in favor of `remotesync`.
This changed the enum values of the keywords (so re-export script and re-build GDNative after this commit).

Closes #7986 .

I still haven't managed to build/install msbuild, so if someone could help testing the Mono build I would appreciate it.

EDIT: Once agreed upon, I'll squash the commits.

### CODE REVIEW

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks, changes look good. If you want to squash commits eventually, I'd suggest to keep the 4th one separate as it's not directly related to the `slave` keyword replacement and deprecation.

**Comment by Faless** [comment]
Squashed

File: modules/mono/csharp_script.cpp [hunk 1]
```diff
@@ -1541,18 +1541,20 @@ MultiplayerAPI::RPCMode CSharpInstance::_member_get_rpc_mode(GDMonoClassMember *
 
 	if (p_member->has_attribute(CACHED_CLASS(RemoteAttribute)))
 		return MultiplayerAPI::RPC_MODE_REMOTE;
-	if (p_member->has_attribute(CACHED_CLASS(SyncAttribute)))
-		return MultiplayerAPI::RPC_MODE_SYNC;
 	if (p_member->has_attribute(CACHED_CLASS(MasterAttribute)))
 		return MultiplayerAPI::RPC_MODE_MASTER;
+	if (p_member->has_attribute(CACHED_CLASS(PuppetAttribute)))
+		return MultiplayerAPI::RPC_MODE_PUPPET;
 	if (p_member->has_attribute(CACHED_CLASS(SlaveAttribute)))
```
Comment (vnen) [inline modules/mono/csharp_script.cpp]: Shouldn't `SlaveAttribute` be changed too?

**Review by vnen — APPROVED** [review]

File: modules/mono/csharp_script.cpp (hunk 1 above)
Comment (akien-mga) [reply modules/mono/csharp_script.cpp]: It's kept as a deprecated synonym for compatibility reasons, we'll remove it for 3.2.

For Mono we could remove it now in theory though, compatibility breakages are expected at least until 3.1.

File: modules/gdscript/gdscript_parser.cpp [hunk 2]
```diff
@@ -4442,7 +4442,8 @@ void GDScriptParser::_parse_class(ClassNode *p_class) {
 				rpc_mode = MultiplayerAPI::RPC_MODE_MASTER;
 				continue;
 			} break;
-			case GDScriptTokenizer::TK_PR_SLAVE: {
+			case GDScriptTokenizer::TK_PR_SLAVE:
```
Comment (akien-mga) [inline modules/gdscript/gdscript_parser.cpp]: Should we add something like:
```cpp
ERR_EXPLAIN("GDScript's 'slave' keyword is deprecated and will be removed in a future release, please replace its uses by 'puppet'.");
WARN_DEPRECATED
```
and fall through to the next case?

Comment (Faless) [reply modules/gdscript/gdscript_parser.cpp]: What about using the beautiful and brand new @vnen warning messages?
deprecated

File: modules/gdscript/gdscript.cpp [hunk 3]
```diff
@@ -1779,10 +1779,11 @@ void GDScriptLanguage::get_reserved_words(List<String> *p_words) const {
 		"remote",
 		"sync",
 		"master",
+		"puppet",
 		"slave",
 		"remotesync",
 		"mastersync",
-		"slavesync",
+		"puppetsync",
```
Comment (LikeLakers2) [inline modules/gdscript/gdscript.cpp]: Wouldn't removing `slavesync` be a breaking change...? Or am I missing something about this change?

Comment (Faless) [reply modules/gdscript/gdscript.cpp]: `slavesync` was never backported to `3.0`, so it was never officially released.

Comment (LikeLakers2) [reply modules/gdscript/gdscript.cpp]: Ah, okay. Nevermind then.

File: modules/gdscript/gdscript_parser.cpp (hunk 2 above)
Comment (akien-mga) [reply modules/gdscript/gdscript_parser.cpp]: :heart:

**Comment by akien-mga** [comment]
Good to merge once CI passes.

**Review by reduz — APPROVED** [review]

**Comment by avril-gh** [comment]
picard
I hope we will not discuss gender of godots logo robot soon ?

**Pull request merged**